src/components/tx-flow/common/TxNonce/index.tsx

```diff
@@ -124,6 +129,20 @@ const TxNonceForm = ({ nonce, recommendedNonce }: { nonce: string; recommendedNo
     formMethods.setValue(TxNonceFormFieldNames.NONCE, recommendedNonce)
   }
 
+  useEffect(() => {
+    let message = ''
+    // Warnings
+    if (Number(nonce) >= safe.nonce + MAX_NONCE_DIFFERENCE) {
+      message = ErrorMessages.NONCE_TOO_FAR
+    }
+
+    if (Number(nonce) > Number(recommendedNonce)) {
+      message = ErrorMessages.NONCE_GT_RECOMMENDED
+    }
+
+    setWarning(message)
+  }, [nonce, recommendedNonce, safe.nonce])
```

Comment:
Why can't this be part of the `validate` call where we also have the rest of the messages?

Reply:
Because in some cases this might be legit what the user wants to do. At least this was like this before the tx flow redesign, IIRC.